sqlcapture/main.go

```diff
@@ -79,16 +79,35 @@ func (d *Driver) Validate(ctx context.Context, req *pc.ValidateRequest) (*pc.Val
 	}
 	defer db.Close(ctx)
 
-	if _, err := DiscoverCatalog(ctx, db); err != nil {
+	discoveredBindings, err := DiscoverCatalog(ctx, db)
+	if err != nil {
 		return nil, err
 	}
 
+	var discoveredStreams []string
+
+	for _, binding := range discoveredBindings {
+		var res Resource
+		if err := pf.UnmarshalStrict(binding.ResourceSpecJson, &res); err != nil {
+			return nil, fmt.Errorf("error parsing resource config: %w", err)
+		}
+		var streamID = JoinStreamID(res.Namespace, res.Stream)
+		discoveredStreams = append(discoveredStreams, streamID)
+	}
+
 	var out []*pc.ValidateResponse_Binding
 	for _, binding := range req.Bindings {
 		var res Resource
 		if err := pf.UnmarshalStrict(binding.ResourceSpecJson, &res); err != nil {
 			return nil, fmt.Errorf("error parsing resource config: %w", err)
 		}
+
+		var streamID = JoinStreamID(res.Namespace, res.Stream)
+
+		if !SliceContains(streamID, discoveredStreams) {
```

Comment:
Unless I've missed some subtlety the whole `discoveredStreams` / `SliceContains` thing could be removed. The `discoveredBindings` map is already keyed by stream ID, so `if _, ok := discoveredBindings[streamID]; ok {` should be equivalent here.

Reply:
@willdonnelly I checked the return value of `DiscoverCatalog` and it's `[]*pc.DiscoverResponse_Binding` so it's not a map, but an array

Reply:
Ah, my mistake. I was thinking of the underlying `DiscoverTables` operation that each database's interface code implements, which returns the map I was thinking of (and then `DiscoverCatalog` implements the whole "translate that into a slice of `pc.DiscoverResponse_Binding` objects with appropriate JSON schemas" step).

Reply:
Thanks, I switched to using that one now 👍🏽